### PR TITLE
Improve language for the last ten years of perl clause for the Lyon Amendment

### PR DESCRIPTION
**Pull request by briandfoy**

### CODE REVIEW

File: lyon-amendment.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ The [Lancaster Consensus](lancaster-consensus.md) wrote:
 
 We have revised this to:
 
-No new release distribution in the Perl toolchain will specify a minimum perl
-prerequisite version (whether configure, build, runtime, or test) that has not
-been available for at least ten years.
+New release distributions in the Perl toolchain will support the last ten
```
Comment (Leont) [inline lyon-amendment.md:17]: I believe the word "support" was deliberately avoided, because of the definition challenges involved

Comment (briandfoy) [reply lyon-amendment.md:17]: What challenges?

Comment (Leont) [reply lyon-amendment.md:17]: What do we commit ourselves to when using the word "support"

Comment (briandfoy) [reply lyon-amendment.md:17]: That the toolchain works with every maintenance release of perl for the past ten years? Isn't the point of this amendment to tell the world which versions of perl the toolchain works with? Is there some concern that the toolchain will not work with recent versions?

Comment (karenetheridge) [reply lyon-amendment.md:17]: I favour the original text -- despite the double negative, it is the most accurate. I would just change "new release distribution" to "release of a distribution".

Comment (kenahoo) [reply lyon-amendment.md:17]: Honestly I can’t even tell what the intention is - to disavow supporting anything older than 10 years, or to commit to supporting everything up to 10 years, or something else?

Also, what happens if Perl 5.x.y is 14 years old, but then 5.x.y+1 is suddenly released, does that bring it into support (I would assume not)?  It’s not accurate to consider releases as a linear set WRT time, but specifying minimum prereqs is a linear cut.  I’m not sure of the wisdom of putting strict date-based language on it when that’s not how it’s ultimately going to be implemented.

Comment (kenahoo) [reply lyon-amendment.md:17]: Regarding the word “support” - would it be better to say “is tested against” or something like that?  Apologies if I’m coming in with too limited an understanding here.

Comment (exodist) [reply lyon-amendment.md:17]: Here is my attempt at wording:

The toolchain group has decided to target a perl release that is at least 10 years old. By target we mean we will make an effort to insure the code works on perls going back 10 years. This is not a guarantee of support, but is an acknowledgement that we believe toolchain modules should be reliable on older version up to a reasonable point, and that point is perl that is 10 years old. Due to a Red Hat support cycle, we do make a one time exception extending this to 12 years as of 2023, 11 years in 2024, and finally 10 years moving forward from 2025 on.

**Comment by Ovid** [comment]
I'm unsure that I'm following all of this, either. I tried to rewrite this so I could understand it.

> Release distributions in the Perl toolchain will not rely on a version of Perl, or related features, which are less than ten years old.
>
> As a one-time exception, the minimum Perl requirement for Red Hat Enterprise Linux and CentOS v7 will not move past v5.16.0 until July 2024, when they leave their maintenance support phase.

Do I have that right? (I'm not suggesting that as text. I'm trying to make sure I understand what's happening here)

**Comment by Ovid** [comment]
Also, I appear to have the ability to merge this pull request, probably because I was much more active in the testing toolchain years ago. I have no intention of abusing this, but someone might want to remove this right from me.

**Comment by Leont** [comment]
rjbs and I already wrote a small [improvement](a0c3f088d408685ee143eff0db885a930a5c8fb2) that we hope solves this discussion.

**Comment by Ovid** [comment]
@Leont I get a 404 on that. Looks like removing my ability to merge also removes my ability to see some things. Not a big deal, in the long run.

**Comment by Leont** [comment]
> @Leont I get a 404 on that. Looks like removing my ability to merge also removes my ability to see some things. Not a big deal, in the long run.

No I just messed up the link, I haven't removed anyone yet. Maybe we should remove some inactive people but that's not super urgent.

File: lyon-amendment.md (hunk 1 above)
Comment (robrwo) [reply lyon-amendment.md:17]: > Also, what happens if Perl 5.x.y is 14 years old, but then 5.x.y+1 is suddenly released, does that bring it into support (I would assume not)?

Perhaps it should be "major release" then?

**Pull request closed without merging**